research_daps/orms/glass.py

```diff
@@ -0,0 +1,202 @@
+# type: ignore
+"""Glass AI ORMs
+
+Assumptions:
+- Organisation names and URL's do not change
+- Sectors do change [TODO]
+- Addresses do change [TODO]
+
+"""
+from sqlalchemy import Table, Column, Integer, String, ForeignKey, Date, Boolean
+from sqlalchemy.orm import relationship, backref
+from research_daps import declarative_base
+
+Base = declarative_base()
+
+# organisation_address = Table(
+#     "organisation_address",
+#     Base.metadata,
+#     Column("org_id", Integer, ForeignKey("organisation.org_id")),
+#     Column("address_id", Integer, ForeignKey("address.address_id")),
+#     # TODO: Active? boolean column indicating whether address is in latest data dump
+#     # TODO: Rank Integer column indicating address rank
+#     # TODO: Date Date column indicating date of latest dump with this relationship
+# )
+class OrganisationAddress(Base):
+    """Association object between organisations and addresses"""
```

Comment:
for all tables, I would set the `__tablename__` class variable so that the table names are prefixed with something like `glass_`

Reply:
Ok, I'd abstracted the boilerplate of setting of `__tablename__` away here: https://github.com/nestauk/research_daps/pull/13/files/54ff408f04d3f8591fb2563463d04b13c54d3bf8#diff-554965363b010d5c38b61d0ffc80e72888e90ff533731b41fe69cc990059a0ff

I could extend this to take a prefix.

Reply:
ah I see, sorry! i didn't see that `Base` was from `research_daps` as I implemented in `ojd_daps`. In the case of `ojd_daps` table prefixes aren't necessary since there is only one project, whereas I guess for `research_daps` there will be multiple projects and so there will be ambiguity that can be cleared up by project-specfici prefixes